File: 2- Even Fibonacci Numbers/Rust/src/main.rs
fn fibo_even_sum(number: u32) -> u32 {
    if number <= 1 {
        return 0;
    } else {
        let mut even_sum = 0;
        let (mut prev_fib_num, mut fib_num) = (1, 2);
        while fib_num <= number {
            if fib_num % 2 == 0 {
                even_sum += fib_num;
            }
            let temp = fib_num;
            fib_num += prev_fib_num;
            prev_fib_num = temp;
        }
        even_sum
    }
}

fn main() {
    let result = fibo_even_sum(4000000);
    println!("{}", result);
}

File: 2- Even Fibonacci Numbers/main.ts
function fiboEvenSum(n: number) {
  if (n <= 1) {
    return 0;
  } else {
    let evenSum: number = 0;
    let prevFibNum: number = 1;
    let fibNum: number = 2;
    for (let i = 2; fibNum <= n; i++) {
      if (fibNum % 2 === 0) {
        evenSum += fibNum;
      }
      [prevFibNum, fibNum] = [fibNum, prevFibNum + fibNum];
    }
    return evenSum;
  }
}
